jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy

```diff
@@ -110,6 +110,19 @@ def step040_install_puppet_code(script_dir, code_deploy, server_era) {
     }
 }
 
+def step045_install_hiera_config(script_dir, code_deploy, server_era) {
+    hiera_config_source_file = (code_deploy["hiera_config_source_file"] == null) ?
+            "" : code_deploy["hiera_config_source_file"]
```

Comment:
Kinda seems like we could check for nil here and just skip the rest of this step from here if it's nil, but I feel like I must be missing something about what you're trying to do.

Also seems like you could check for the existence of the file here and if it doesn't exist, fail.  I've probably been inconsistent about doing checks like that from the groovy code vs. the shell code or beaker scripts, but my gut is that it'd be better to have as much of that in the groovy as possible?  WDYT?

Anyway, if there is some reason not to skip this step if that value is nil, then you can ignore all these comments and this is probably fine how it is.